ansible_rulebook/action/base_action.py

```diff
@@ -0,0 +1,90 @@
+#  Copyright 2023 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+import uuid
+from typing import Dict
+
+from ansible_rulebook.conf import settings
+from ansible_rulebook.event_filter.insert_meta_info import main as insert_meta
+from ansible_rulebook.util import run_at
+
+from .control import Control
+from .metadata import Metadata
+
+KEY_EDA_VARS = "ansible_eda"
+INTERNAL_ACTION_STATUS = "successful"
+
+
+class BaseAction:
```

Comment:
Use delegation instead of inheritance for code reuse.   https://en.wikipedia.org/wiki/Delegation_pattern